src/cli/subcommand/export.rs

```diff
@@ -0,0 +1,291 @@
+    InvalidXdgDataDirs(Vec<PathBuf>),
+
+    #[error("Some of the paths for PATH are not valid, due to an illegal character, like a space or colon.")]
+    InvalidPathDirs(Vec<PathBuf>),
+
+    #[error("Some of the paths for MANPATH are not valid, due to an illegal character, like a space or colon.")]
+    InvalidManPathDirs(Vec<PathBuf>),
+}
+
+/**
+Emit all the environment variables that should be set to use Nix.
+
+Safety note: environment variables and values can contain any bytes except
+for a null byte. This includes newlines and spaces, which requires careful
+handling.
+
+In `space-newline-separated` mode, `nix-installer` guarantees it will:
+
+  * only emit keys that are alphanumeric with underscores,
+  * only emit values without newlines
+
+and will refuse to emit any output to stdout if the variables and values
+would violate these safety rules.
+
+In `null-separated` mode, `nix-installer` emits data in this format:
+
+  KEYNAME\0VALUE\0KEYNAME\0VALUE\0
+
+*/
+#[derive(Debug, Parser)]
+#[command(args_conflicts_with_subcommands = true)]
+pub struct Export {
+    #[clap(long)]
+    format: ExportFormat,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, clap::ValueEnum)]
+enum ExportFormat {
+    NullSeparated,
+    SpaceNewlineSeparated,
+}
+
+#[async_trait::async_trait]
+impl CommandExecute for Export {
+    #[tracing::instrument(level = "trace", skip_all)]
+    async fn execute(self) -> eyre::Result<ExitCode> {
+        let env = match calculate_environment() {
+            Err(Error::AlreadyRun) => {
+                debug!("Already set the environment vars, not doing it again.");
+                return Ok(ExitCode::SUCCESS);
+            },
+            Err(e) => {
+                tracing::info!(
+                    "Error calculating the environment variables required to enable Nix: {:?}",
+                    e
+                );
+                return Err(e.into());
+            },
+
+            Ok(env) => env,
+        };
+
+        let mut out = BufWriter::new(stdout());
+
+        match self.format {
+            ExportFormat::NullSeparated => {
+                debug!("Emitting null separated fields");
+
+                for (key, value) in env.into_iter() {
+                    out.write_all(key.as_bytes())?;
+                    out.write_all(&[b'\0'])?;
+                    out.write_all(&value.into_vec())?;
+                    out.write_all(&[b'\0'])?;
+                }
+            },
+            ExportFormat::SpaceNewlineSeparated => {
+                debug!("Emitting space/newline separated fields");
+
+                let mut validated_envs = HashMap::new();
+                for (key, value) in env.into_iter() {
+                    if !key.chars().all(|char| {
+                        "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_"
+                            .contains(char)
+                    }) {
+                        warn!("Key {} has an invalid character that isn't a-zA-Z_", key);
```

Comment:
Characters 0–9 are allowed, but not mentioned in the warning.

Reply:
Ah, thank you for noticing! Also, I really appreciate you coming and doing a review =)!